### ISSUE TITLE
RW-TURP - Web Application Acceptance validation on Responsible Party (SS) (On PROD)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#13578]
## Don't save approved attempt on Web Application Acceptance on TURP permits if Responsible Party bond or insurance is expired.

**Acceptance Criteria** 

When user makes the Accepted attempt on Web Application Acceptance process in RW folders of subtype TURP, system must perform the following function:
- Check to see if the folder has a Responsible Party people record added to it. 
    - If Responsible Party people record doesn't exist on the folder, do not run the validation, just allow the attempt to save. 
- If there is a Responsible Party on the folder, check the People Info of the Responsible Party People Record for infos 'Bond Expiration Date'  (58085) and 'Insurance Expiration Date' (75350)
- If either of these People Infos have a date that is in the past, or a blank value, system must display a hard stop message stating Bond and/or Insurance information is out of date.

TURP permits require current/valid insurance and bond. We'd like the system to validate both before issuing a Temporary Use of Right of Way Permit. If either insurance/bond are expired, do not issue the permit.
- _New issue for #1802_

----
**Notes for Scoping**
- Does this only apply to RW-TURP? 
- Where/when should the validation be triggered? 
- How does this apply to permit extensions? 
- It's worth considering if this feature should/could warn the user that the insurance/bond expires before the permit expires.

**Comment by TracyLinder** [comment]
### Scoping during ROW Friday review 9/15/23

- Sree & Tracy suggest putting the validation trigger on a folder process instead of in an issuance procedure, due to issues with the activate upon payment feature. 
- ~The last process PLA uses to indicate the review determination is 50520, TURP Review. The **approved** attempt should trigger the validation.~ 
    - There is concern that validating on this process would cause issues. Kim would rather use the **accepted** attempt on the Web Application Acceptance Process.

**Comment by vschandramouli** [comment]
Updated AC, moving to Backlog.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  I committed the developed code for this ticket [a72384f1de44](http://coaamandadevops02.austintexas.gov/R5:a72384f1de448cbda5bcf92161033a9715105b8d). Please approve. Thanks.

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been sent to the DevOps CM Team to be deployed to DEV and TEST under **SCTASK0362218**

**Comment by SubhaShroff** [comment]
Thank you for raising the request @Nadin-Nader.
It looks deployed I will be testing tomorrow.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @KyleErich This is tested on TEST. 
You can use PeopleRSN - 5831040 for testing on TEST. It has Bond/Insurance expiration dates with future dates.

**Comment by TracyLinder** [comment]
@vschandramouli when you're facilitating UAT for this, can you please inform Kyle that according to the current build - if there is no Responsible Party the system won't do a validation. Will this mean ROW might inadvertently issue a TURP without an insurance/bond validation.

**Comment by Nadin-Nader** [comment]
Update from Kyle:

> 13578: UAT postponed. I want to verify with the team at the Friday meeting if we're okay with the fact that the validation won't run if there is no Responsible Party on a RW-TURP. UAT shows it is working as planned but I want to ensure that the plan is correct.

**Comment by TracyLinder** [comment]
@SubhaShroff please remove the validation that the TURP must have a responsible party people type. I updated the AC.

**Comment by Nadin-Nader** [comment]
This has been sent to the DevOps CM Team to be deployed to DEV and TEST under **SCTASK0365917**

**Comment by Nadin-Nader** [comment]
@SubhaShroff this has been deployed. Please UAT

**Comment by SubhaShroff** [comment]
Thanks @Nadin-Nader.
Tested on TEST for RSNs - 13107132, 13199389. Ready for UAT.
CC: @KyleErich

**Comment by KyleErich** [comment]
When we tested with no value or future value, it was successful. We were not able to create a record for testing with an expiration date in the past due to permissions issues. Before approving UAT, we want to test this scenario as well.
@vschandramouli

**Comment by vschandramouli** [comment]
@SubhaShroff can you help set up Responsible Party People records in TEST that has bond and insurance expiry dates in the past so we can test this last scenario? 

CC @Nadin-Nader

**Comment by Nadin-Nader** [comment]
Waiting on CTM to give permissions in order to UAT

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @vschandramouli @KyleErich 
for now with Anu's help, I got the people info  updated with past dates for PeopleRSN - 5831078 on TEST. You can use this Responsible Party type to test on TEST.
There are other people records too:
5831077 - bond expiry date in FUTURE, Insurance expiry date in PAST.
5831040 - Both future dates.

Thanks.

**Comment by KyleErich** [comment]
@Nadin-Nader @SubhaShroff 
UAT approved on Test. Thanks.

**Comment by SubhaShroff** [comment]
Thank you Kyle.
@Nadin-Nader please create a PROD promotion request. Here is the Deployment Plan for it.
[13578 - a72384f1de44 - c2c2d2134c68 - PROD Deployment Plan.docx](https://github.com/cityofaustin/atd-data-tech/files/13208674/13578.-.a72384f1de44.-.c2c2d2134c68.-.PROD.Deployment.Plan.docx)

**Comment by Nadin-Nader** [comment]
@SubhaShroff  this has been sent to the DevOps CM Team to be deployed to PROD under **SCTASK0367626**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  This is deployed in PROD. Ready for users. Thanks.

**Comment by Nadin-Nader** [comment]
Sent Kyle a message via Teams that this is ready for UAT on PROD. Waiting on Response

**Comment by Nadin-Nader** [comment]
@SubhaShroff here is Kyle's message:

13578: UAT rejected. I was able to issue FolderRSN 13236406 even though my Responsible Party listing has null Bond and Insurance Expiration Dates.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @KyleErich Though the title of this ticket says issuance, the AC talks about WAA process Acceptance, only. I haven't focused on the title. It might be tested for WAA process but not permit issue when this ticket was approved for PROD. Please let me know if this validation should work for ISSUE as well.

**Comment by SubhaShroff** [comment]
@Nadin-Nader As per our discussion on Friday 11/17/2023, Tracy limited this ticket to WAA process Accepted attempt. And she has created another ticket to do the validation with Issuance. 
So, this is on PROD and can be tested for WAA validation.

**Comment by Nadin-Nader** [comment]
Update: Waiting on #14684

**Comment by TracyLinder** [comment]
@SubhaShroff can you please confirm if this is in PROD or not? It's unclear.

**Comment by TracyLinder** [comment]
Subha says this is in PROD - we need confirmation that it is working on PROD. 
@KyleErich are you able to validate this is working in PROD?

**Comment by TracyLinder** [comment]
We think this is working in PROD. OK to close.
